staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -569,6 +605,50 @@ func (s *sharedIndexInformer) HandleDeltas(obj interface{}) error {
 	return nil
 }
 
+// IsStarted reports whether the informer has already been started
+func (s *sharedIndexInformer) IsStarted() bool {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+	return s.started
+}
+
+// IsStopped reports whether the informer has already been stopped
+func (s *sharedIndexInformer) IsStopped() bool {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+	return s.stopped
+}
+
+// EventHandlerCount reports whether the informer still has registered
+// event handlers
+func (s *sharedIndexInformer) EventHandlerCount() int {
+	s.startedLock.Lock()
+	defer s.startedLock.Unlock()
+	return len(s.processor.listeners)
+}
```

Comment:
They are intended for a informer management layer on top, that manages the handler and watches.
For sure, this must be synchronized again. You are right that the naked usage of those methods bear the potential for race conditions. But without them it would not be possible to offer higher layers outside the cache package.

Reply:
> it's not clear to me how they are going to be used (i.e. the informer management layer on top). suggest to make it private for this PR, so that it is focused on supporting removal of event handler only.

agreed

Reply:
I'm just doing a pre-review skim of this PR, this comment (make these three private) doesn't seem to have been applied.

I'm not sure what "higher layers" might need to do but we can enable it specifically when we figure it out.